packages/enhanced/src/lib/sharing/ConsumeSharedPlugin.ts

```diff
@@ -316,6 +320,8 @@ class ConsumeSharedPlugin {
               chunk,
               new ConsumeSharedRuntimeModule(set),
             );
+            // FIXME: need to remove webpack internal inject ShareRuntimeModule, otherwise there will be two ShareRuntimeModule
+						compilation.addRuntimeModule(chunk, new ShareRuntimeModule());
```

Comment:
@ScriptedAlchemy  Now dist will have two ShareRuntimeModule , and i control the enhanced/ShareRuntimeModule execute after original ShareRuntimeModule .

But the original ShareRuntimeModule still there . Do you have any idea to delete it ?

Reply:
IS the original share runtime module not under our control? Why is the original one in webpck getting referenced?
I have experienced this issue when applying module federation or container plugin more than once in a build. Dirty workaround was i made up new runtime requirement that i add to track if someone has already added it to chunk. 

The other option is to call getRuntimeModule off chunkGraph and unshift / remove connection or dedupe it. 

I think if you use same tap("name" as the plugin, it can overwrite sometimes, otherwise you could use interceptor to tap and prevent webpack from using plugin or adding it. But if there is no good hook this is hard

Reply:
This is hacky, but does work. 

```js
class ModuleInfoRuntimePlugin {
  apply(compiler: Compiler) {
    compiler.hooks.thisCompilation.tap(
      'ModuleInfoPlugin',
      (compilation: Compilation) => {
        compilation.hooks.additionalTreeRuntimeRequirements.tap(
          'ModuleInfoRuntimeModule',
          (chunk, set) => {
            //workaround
            if (set.has('__webpack_require__.federation')) return;
            //TODO: this should be added as a handler, so one can push requirements in and we call the for tap to add module once.
            set.add('__webpack_require__.federation');

            set.add(RuntimeGlobals.global);

            set.add(RuntimeGlobals.runtimeId);
            compilation.addRuntimeModule(chunk, new ModuleInfoRuntimeModule());
          },
        );
      },
    );
  }
}```

Reply:
Howver in this case, i still have access to the class so i can control duplication since its not replacement of other module we cant control. 

`compilation.chunkGraph.getChunkRuntimeModulesInOrder(chunk);` - but we need to get this hook in later stage once runtime modules have been added

Reply:
From there - we could see module identifier. like webpack/sharing/init, and filter out webpacks in place of enhanced/sharing/init runtime module?